src/lib/alloc.c

```diff
@@ -560,30 +560,41 @@ void *_malloc(int zone, uint32_t caps, size_t bytes)
 	uint32_t flags;
 	void *ptr = NULL;
 
-	spin_lock_irq(&memmap.lock, flags);
+	if (bytes >= HEAP_BUFFER_BLOCK_SIZE) {
+		ptr = _balloc(zone, caps, bytes);
```

Comment:
Best to just return _balloc(zone, caps, bytes) here as it saves the extra indentation below.